src/users-api.ts

```diff
@@ -261,6 +268,9 @@ const postMetadata = function (req, res, next) {
         err,
         reply
       });
+    } else if (key === 'email' && req.params.user.email &&
+      !emails.isGuestEmail(value) && !emails.isNoEmail(value)) {
+      emailConfirmation?.sendEmailConfirmation(params, params.username, value, true);
     }
     res.send({ ok: !err });
```

Comment:
The change that resolves this basically does:
```ts
if (key === 'email') {
  // ...
  needEmailConfirmation = true;
}
res.send({ ok: !err, needEmailConfirmation });
```

Unfortunately, the point of this is not to check that we updated our email address (because we already know that), but to know if the confirmation email was sent or not.

In `emailConfirmation.sendEmailConfirmation` we are only sending an email if this email was never confirmed by the user, the value of `needEmailConfirmation` should only be true when the email was sent.

(I'm fixing this now)